### PR TITLE
report-ids endpoint

### PR DESCRIPTION
**Pull request by rickmarry**
### :nut_and_bolt: Description
This endpoint returns the list of compliance run-ids for the given input filter.  All of the compliance supported filters are supported and both start_time as well as end_time are used when filtering the list.

This is useful for getting all of the runids that happened inside of a timespan.

### :white_check_mark: Checklist

- [ ] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [ ] Code actually executed?
- [ ] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/automate-gateway/api/compliance/reporting/reporting.proto [hunk 1]
```diff
@@ -30,6 +30,21 @@ service ReportingService {
 			action: "compliance:reports:list"
 		};
 	};
+	// should cover /reportids
+    rpc ListReportIds(Query) returns (ReportIds) {
```
Comment (vjeffrey) [inline components/automate-gateway/api/compliance/reporting/reporting.proto]: can you add a comment here noting that this endpoint is *not* limited by the 10k es pagination thing

File: components/compliance-service/api/reporting/server/server.go [hunk 2]
```diff
@@ -63,6 +63,48 @@ func (srv *Server) ListReports(ctx context.Context, in *reporting.Query) (*repor
 	return &reports, nil
 }
 
+// ListReports returns a list of reports based on query
+func (srv *Server) ListReportIds(ctx context.Context, in *reporting.Query) (*reporting.ReportIds, error) {
```
Comment (vjeffrey) [inline components/compliance-service/api/reporting/server/server.go]: i know the pr is still marked as wip...but i assume you'll be adding a test or two for this new fancy endpoint?

Comment (rickmarry) [reply components/compliance-service/api/reporting/server/server.go]: yep.. i got the tests in now.. will take this out of WIP

File: components/automate-gateway/api/compliance/reporting/reporting.proto (hunk 1 above)
Comment (rickmarry) [reply components/automate-gateway/api/compliance/reporting/reporting.proto]: done

Comment (rickmarry) [reply components/automate-gateway/api/compliance/reporting/reporting.proto]: done

**Pull request merged**